org.civicrm.reqangular/test/services/api.spec.js

```diff
@@ -99,26 +103,84 @@ define([
         });
       });
 
+      describe('caching', function () {
+        var sampleResult = 'sample result';
+        var sampleParams = {};
+
+        beforeEach(function () {
+          spyOn($cacheFactory.get('$http'), 'remove').and.callThrough();
+        });
+
+        describe('when "returnCachedData" parameter is not passed', function () {
+          beforeEach(function () {
+            expectAndSendGET(sampleResult, sampleParams);
+            $rootScope.$digest();
+          });
+
+          it('does not flush cache before HTTP call', function () {
+            expect($cacheFactory.get('$http').remove).not.toHaveBeenCalled();
+          });
+
+          it('caches data by default', function () {
+            expect($cacheFactory.get('$http').info().size).toBe(1);
+          });
+        });
+
+        describe('when "returnCachedData" parameter passed as TRUE', function () {
+          beforeEach(function () {
+            expectAndSendGET(sampleResult, sampleParams, true);
+            $rootScope.$digest();
+          });
+
+          it('does not flush cache before HTTP call', function () {
+            expect($cacheFactory.get('$http').remove).not.toHaveBeenCalled();
+          });
+
+          it('caches data', function () {
+            expect($cacheFactory.get('$http').info().size).toBe(1);
+          });
+        });
+
+        describe('when "returnCachedData" parameter passed as FALSE', function () {
+          beforeEach(function () {
+            expectAndSendGET(sampleResult, sampleParams, false);
+            $rootScope.$digest();
+          });
+
+          it('flushes cache before HTTP call', function () {
+            expect($cacheFactory.get('$http').remove).toHaveBeenCalledWith(
+              '/civicrm/ajax/rest?action=action&entity=entity&json=%7B%22options%22:%7B%22limit%22:0%7D%7D&sequential=1');
```

Comment:
it's not clear at all where you got this string. Either you put in a well-named variable or maybe you can have it returned by `expectAndSendGET()`. Anyway this needs to be improved

Reply:
Done and added a comment